InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Exceptions.Modify.cs

```diff
@@ -0,0 +1,228 @@
+using Xunit;
+using FluentAssertions;
+using Microsoft.EntityFrameworkCore;
+
+namespace InternTrack.Core.Api.Tests.Unit.Services.Foundations.Interns
+{
+    public partial class InternServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowCriticalDependencyExceptionOnModifyIfSqlErrorOccursAndLogItAsync()
+        {
+            // given
+            DateTimeOffset datetime = GetRandomDateTime();
+            Intern randomIntern = CreateRandomIntern(datetime);
+            SqlException sqlException = GetSqlException();
+
+            var failedInternStorageException =
+                new FailedInternStorageException(sqlException);
+
+            var expectedInternDependencyExcetpion =
+                new InternDependencyException(failedInternStorageException);
+
+            this.dateTimeBrokerMock.Setup(broker =>
+                broker.GetCurrentDateTimeOffset())
+                    .Throws(sqlException);
+
+            // when
+            ValueTask<Intern> modifyInternTask =
+                this.internService.ModifyInternAsync(randomIntern);
+
+            InternDependencyException actualInternDependecyException =
+                await Assert.ThrowsAsync<InternDependencyException>(
+                    modifyInternTask.AsTask);
+
+            // then
+            actualInternDependecyException.Should().BeEquivalentTo(
+                expectedInternDependencyExcetpion);
+
+            this.dateTimeBrokerMock.Verify(broker =>
+                broker.GetCurrentDateTimeOffset(),
+                    Times.Once());
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogCritical(It.Is(SameExceptionsAs(
+                    expectedInternDependencyExcetpion))),
+                        Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectInternByIdAsync(randomIntern.Id),
+                    Times.Never());
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowDependencyExceptionOnModifyIfDatabaseUpdateExceptionOccursAndLogItAsync()
+        {
+            // given
+            DateTimeOffset datetime = GetRandomDateTime();
+            Intern randomIntern = CreateRandomIntern(datetime);
+            randomIntern.CreatedDate = datetime;
+            randomIntern.UpdatedDate = randomIntern.CreatedDate.AddMinutes(GetRandomNumber());
+
+            var databaseUpdateException = new DbUpdateException();
+
+            var failedInternStorageException =
+                new FailedInternStorageException(databaseUpdateException);
+
+            var expectedInternDependencyException =
+                new InternDependencyException(failedInternStorageException);
+
+            this.dateTimeBrokerMock.Setup(broker =>
+                broker.GetCurrentDateTimeOffset())
```

Comment:
You can make datetimebroker to throw the exception to minimize the code and improve readibility. Just as the test above